src/db.rs

```diff
@@ -1083,48 +1085,51 @@ impl CommitOverlay {
 		self.btree_indexed.get(key).map(|(_, v)| v.as_ref())
 	}
 
-	pub fn btree_next(
-		&self,
-		last_key: &Option<Vec<u8>>,
-		from_seek: bool,
-	) -> Option<(Value, Option<Value>)> {
-		let key = if let Some(key) = last_key.as_ref() {
-			key
-		} else {
-			return self.btree_indexed.iter().next().map(|(k, (_, v))| (k.clone(), v.clone()))
-		};
-
-		let mut iter = self.btree_indexed.range::<Vec<u8>, _>(key..);
-
-		let (k, v) = if let Some((k, (_, v))) = iter.next() { (k, v) } else { return None };
-
-		if from_seek || k != key {
-			return Some((k.clone(), v.clone()))
+	pub fn btree_next(&self, last_key: &crate::btree::LastKey) -> Option<(Value, Option<Value>)> {
```

Comment:
I believe might be possible to return slices instead of `Value` here but it's likely to not be very useful anyway...

Reply:
I will keep it as is (the result is return as Vec<u8> since it can come from the file backend too, could return some Cow, but it will change all api so would be a bit overkill).

Reply:
Ok! Make sense.